src/main/java/io/supertokens/useridmapping/UserIdMapping.java

```diff
@@ -218,77 +214,66 @@ public static boolean updateOrDeleteExternalUserIdInfo(Main main,
                                                            String userId, UserIdType userIdType,
                                                            @Nullable String externalUserIdInfo)
             throws StorageQueryException {
-        try {
-            return updateOrDeleteExternalUserIdInfo(new TenantIdentifier(null, null, null), main, userId, userIdType,
-                    externalUserIdInfo);
-        } catch (TenantOrAppNotFoundException e) {
-            throw new IllegalStateException(e);
-        }
+        Storage storage = StorageLayer.getStorage(main);
+        return updateOrDeleteExternalUserIdInfo(new AppIdentifier(null, null, storage),
+                userId, userIdType, externalUserIdInfo);
     }
 
-    public static HashMap<String, String> getUserIdMappingForSuperTokensUserIds(TenantIdentifier tenantIdentifier,
-                                                                                Main main,
+    public static HashMap<String, String> getUserIdMappingForSuperTokensUserIds(AppIdentifier appIdentifier,
                                                                                 ArrayList<String> userIds)
-            throws StorageQueryException, TenantOrAppNotFoundException {
-        return StorageLayer.getUserIdMappingStorage(tenantIdentifier, main)
-                .getUserIdMappingForSuperTokensIds(tenantIdentifier.toAppIdentifier(), userIds);
+            throws StorageQueryException {
+        // This is tenant specific, so just use the storage from the appIdentifier
+        return appIdentifier.getUserIdMappingStorage().getUserIdMappingForSuperTokensIds(appIdentifier, userIds);
     }
```

Comment:
why is this tenant specific?

Reply:
It's called only from "/users" API which lists all users in a tenant.

Reply:
But now you have converted it to being appIdentifier - so im confused.

Reply:
Have made changes so that it makes more sense now.